src/syndrome_decoder.jl

```diff
@@ -96,5 +98,89 @@ function syndrome_decode(pcm, pcmT, syndrome, max_iters, channel_probs, b2c, c2b
   end
 
   return Bool.(error), converged
+end
+
+# In-place variant
+# TODO: Currently very slow. Need to improve this
```

Comment:
I suspect this would be much faster if it can be written in a more "linear algebra" fassion. Each of the belief prop iterations should look very similar to a matrix multiplication (just how each layer of a neural net looks like a matrix multiplication). No need to fix that in the current PR, just suggesting potential future improvement.

Reply:
Thanks, will be taking this issue up next.